pages/blogs.tsx

```diff
@@ -1,14 +1,89 @@
+import commerce from '@lib/api/commerce';
+import { GetStaticPropsContext } from 'next';
 import { Layout } from 'src/components/common';
-import { BlogsList, FeaturedCardBlog, BlogHeading, BlogBreadCrumb } from 'src/components/modules/blogs';
+import { BlogCardProps } from 'src/components/common/CardBlog/CardBlog';
+import { BlogBreadCrumb, BlogHeading, BlogsList, FeaturedCardBlog } from 'src/components/modules/blogs';
+import { DEFAULT_BLOG_PAGE_SIZE } from "src/utils/constanst.utils";
+import { getAllPromies } from 'src/utils/funtion.utils';
+import { PromiseWithKey } from 'src/utils/types.utils';
 
-export default function BlogsPage() {
+interface Props {
+    blogsResult: { blogs?: BlogCardProps[],featuredBlog?: BlogCardProps[] },
+    totalItems: number
+}
+export default function BlogsPage({blogsResult,totalItems}:Props) {
+    let date = new Date(blogsResult.featuredBlog?.[0]?.createdAt ?? '' );
+    let fullDate = date.toLocaleString('en-us', { month: 'long' }) + " " + date.getDate()+","+date.getFullYear();
+  
     return(
         <>
             <BlogBreadCrumb />
             <BlogHeading />
-            <FeaturedCardBlog />
-            <BlogsList />
+            <FeaturedCardBlog 
+            title={blogsResult.featuredBlog?.[0]?.title} 
+            slug={blogsResult.featuredBlog?.[0]?.slug} 
+            imgSrc={blogsResult.featuredBlog?.[0]?.imageSrc ?? ''}
+            content={blogsResult.featuredBlog?.[0]?.description}
+            imgAuthor={blogsResult.featuredBlog?.[0]?.authorAvatarAsset}
+            authorName={blogsResult.featuredBlog?.[0]?.authorName}
+            date={fullDate}
+            />
+            <BlogsList blogList={blogsResult.blogs} total={totalItems} idFeatured={blogsResult.featuredBlog?.[0]?.id} />
         </>
     )
 }
-BlogsPage.Layout = Layout
\ No newline at end of file
+
+
+export async function getStaticProps({
+  preview,
+  locale,
+  locales,
+}: GetStaticPropsContext) {
+  const config = { locale, locales }
+  let promisesWithKey = [] as PromiseWithKey[]
+  let props = {} as any;
+
+
+  const {featuredBlogs} = await commerce.getFeaturedBlog({
+    variables: {
+      take: DEFAULT_BLOG_PAGE_SIZE
```

Comment:
We only need 1 featured blog
```suggestion
      take: 1
```